src/Orleans.Transactions/DistributedTM/Factories/NamedTransactionalStateStorageFactory.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System;
+using Microsoft.Extensions.DependencyInjection;
+using Orleans.Runtime;
+using Orleans.Transactions.Abstractions;
+using Orleans.Storage;
+using Microsoft.Extensions.Logging;
+
+namespace Orleans.Transactions.DistributedTM
+{
+    public class NamedTransactionalStateStorageFactory : INamedTransactionalStateStorageFactory
+    {
+        private readonly IGrainActivationContext context;
+        public NamedTransactionalStateStorageFactory(IGrainActivationContext context)
+        {
+            this.context = context;
+        }
+        public ITransactionalStateStorage<TState> Create<TState>(string storageName)
+            where TState : class, new()
+        {
+            // Try to get ITransactionalStateStorage from factory
+            ITransactionalStateStorageFactory factory = string.IsNullOrEmpty(storageName)
+                ? this.context.ActivationServices.GetService<ITransactionalStateStorageFactory>()
+                : this.context.ActivationServices.GetServiceByName<ITransactionalStateStorageFactory>(storageName);
+            if (factory != null) return factory.Create<TState>();
+
+            // Else try to get storage provider and wrap it
+            IStorageProvider storageProvider = string.IsNullOrEmpty(storageName)
```

Comment:
I think this should be updated to use IGrainStorage, `TransactionalStateStorageProviderWrapper` doesn't take IStorageProvider anymore